qemu/tests/bridge_qinq.py

```diff
@@ -0,0 +1,318 @@
+        :param enable_logging: whether to dump tcpdump results during test
+        """
+        get_tcpdump_log_cmd = params["get_tcpdump_log_cmd"] % iface_name
+        tcpdump_content = session.cmd_output(get_tcpdump_log_cmd, timeout=300).strip()
+        lines = tcpdump_content.split("\n")
+        sum = 0
+        for i in range(len(lines)):
+            if enable_logging:
+                logging.info("line %s: %s" % (i, lines[i]))
+            if not ethertype2:
+                if "ICMP echo re" in lines[i] and \
+                        ethertype in lines[i-1]:
+                    sum += 1
+                    if vlan_tag and vlan_tag not in lines[i-1]:
+                        if "too much work for irq" in lines[i-1]:
+                            continue
+                        else:
+                            test.fail("in %s tcpdump log, there should be vlan "
+                                      "tag %s" % (iface_name, vlan_tag))
+                    elif not vlan_tag:
+                        if "vlan" in lines[i-1]:
+                            test.fail("in %s tcpdump log, there should not be "
+                                      "vlan tag" % iface_name)
+            else:
+                if "ICMP echo re" in lines[i] and \
+                        ethertype in lines[i-1] and \
+                        ethertype2 in lines[i-1]:
+                    sum += 1
+                    if vlan_tag not in lines[i-1] or \
+                            vlan_tag2 not in lines[i-1]:
+                        if "too much work for irq" in lines[i-1]:
+                            continue
+                        else:
+                            test.fail("in %s tcpdump log, there should be vlan "
+                                      "tag %s" % (iface_name, vlan_tag))
+        if sum == 0:
+            test.fail("in %s tcpdump log, ethertype is not %s" % (iface_name,
+                                                                  ethertype))
+
+    def compare_host_guest_md5sum(name):
+        """
+        Compare md5 value of file on host and guest
+
+        :param name: file name
+
+        """
+        txt = "Comparing md5sum of the file %s on guest and host" % name
+        error_context.context(txt, logging.info)
+        host_result = crypto.hash_file(name, algorithm="md5")
+        try:
+            output = session.cmd_output("md5sum %s" % name)
+            guest_result = re.findall(r"\w+", output)[0]
+        except IndexError:
+            logging.error("Could not get file md5sum in guest")
+            return False
+        logging.debug("md5sum: guest(%s), host(%s)", guest_result, host_result)
+        return guest_result == host_result
+
+    login_timeout = int(params.get("login_timeout", "600"))
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login(timeout=login_timeout)
+    guest_qinq_dir = params["guest_qinq_dir"]
```

Comment:
Missing this parameter.

Reply:
Missing it cause:
 (1/1) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.bridge_qinq.q35: SKIP: Mandatory parameter 'guest_qinq_dir' is missing. Check your cfg files for typos/mistakes

Reply:
@pezhang @PaulYuuu Updated. Please help review again. Thanks.